challenge_4/python/bryantpq/tree.py

```diff
@@ -0,0 +1,72 @@
+class Node:
+    def __init__(self, value):
+        self.value = value
+        self.l = None
+        self.r = None
+
+class BinaryTree:
+    def __init__(self):
+        self.root = None
+
+    def draw(self):
+        '''Prints a preorder traversal of the tree'''
+        self._draw(self.root)
+        print()
+
+    def _draw(self, node):
+        print("(", end="")
+        if node != None:
+            print(str(node.value) + ", ", end="")
+            self._draw(node.l)
+            print(", ", end="")
+            self._draw(node.r)
+        print(")", end="")
+
+
+    def invert(self):
+        self.root = self._invert(self.root)
+
+
+    def _invert(self, node):
+        # find lowest point where nodes can be swapped
+        # swap nodes
+        if not (node.l.l == None and node.l.r == None): # node.left is terminal/leaf
```

Comment:
This is a bit of a mouthful. 

In essense you're checking $not (not node.l.r and not node.l.r)$. This can be boiled down to $node.l.l or node.l.r$ which doesn't actually check if `node.left` is a leaf or not.  In fact, it checks if `node.left` isn't a leaf. Also, keep in mind that if `node.l` is None, you're going to get an error.